### PR TITLE
the book list has been created

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary

- First structure of library.
- Render the book list.
- Fix styling bugs.

#### Galaxy S (or similar)

image

#### Pixel 2/3 (or similar)

image

#### Pixel 3XL (or similar)

image

#### iPhone SE (or similar)

image

#### iPhone 6/7/8 (or similar)

image

#### iPhone 11/12 Base/Pro/Pro Max (or similar)

image

## Trello Card

https://trello.com/c/sg6qQTIY/10-n-61-componente-libro
https://trello.com/c/4OuZ3JPl/9-n-62-lista-de-libros

### CODE REVIEW

File: wbooks/src/app/screens/BookData/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  id: number;
+  author: string;
+  title: string;
+  genre: string;
+  publisher: string;
+  year: string;
+  imageUrl: string | null;
+}
+
+const BookData: ListRenderItem<Props> = ({ item }) => (
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookData/index.tsx]: Book Data it's not a screen, It's a component of your Library or a component of the app (if is utilize in other screens), but definitely not a screen. Move it on to components folder inside of your library screen.

File: wbooks/src/app/screens/BookData/styles.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  imageBook: {
+    height: 80,
+    width: 60,
+    resizeMode: 'stretch',
+    marginRight: 20
+  },
+  title: {
+    fontSize: 18,
+    fontWeight: 'bold',
+    width: 200
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookData/styles.ts]: don´t fix the width. Allow the component grown, use padding to separate with the border of the father.

File: wbooks/src/app/screens/BookData/styles.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  author: {
+    fontSize: 16
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookData/styles.ts]: Set a fontSize file (or fonts, your choose) with all constant so you can use something like "fontSize: size.normal". You can see why this approach is an advantage, right?.

File: wbooks/src/app/screens/BookData/styles.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+export default StyleSheet.create<Styles>({
+  container: {
+    flex: 1,
+    marginTop: 35
+  },
+  bookCard: {
+    backgroundColor: COLORS.white,
+    borderRadius: 10,
+    padding: 30,
+    marginBottom: 10,
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookData/styles.ts]: use padding on the parent, if you were looking to separate parent and child.

File: wbooks/src/app/screens/BookData/index.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  <TouchableOpacity style={styles.bookCard}>
+    <View style={styles.dataContainer}>
+      <Image source={item.imageUrl ? { uri: item.imageUrl } : bcInicio} style={styles.imageBook} />
```
Comment (mcolombo87) [inline wbooks/src/app/screens/BookData/index.tsx]: deconstruct all item to imageUrl, title, author, etc.

**Review by mcolombo87 — CHANGES_REQUESTED** [review]
Don't merge in one PR two different cards please.

**Review by guidoprinc — DISMISSED** [review]
check if maxi brings new comments

File: wbooks/src/app/screens/Library/components/BookData/index.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  id: number;
+  author: string;
+  title: string;
+  genre: string;
+  publisher: string;
+  year: string;
+  imageUrl: string | null;
+}
+
+const BookData: ListRenderItem<Props> = ({ item }) => {
```
Comment (mcolombo87) [inline wbooks/src/app/screens/Library/components/BookData/index.tsx]: You are typing this as "ListRenderItem" so, this components can be used only inside of a renderItem. That is fine but you defined this "BookData" as a componente of Library (for all screen inside of Library) and perhaps only is a component of BookList.

File: wbooks/src/app/screens/Library/screens/BookList/index.tsx [hunk 7]
```diff
@@ -2,11 +2,13 @@ import React from 'react';
 import { View, FlatList } from 'react-native';
 import { BOOKS_MOCK } from '@constants/mockBooks';
 
-import BookData from '../BookData';
+import BookData from '../../components/BookData';
```
Comment (mcolombo87) [inline wbooks/src/app/screens/Library/screens/BookList/index.tsx]: use Alias.

File: wbooks/src/constants/fonts.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export const FONTS = {
```
Comment (mcolombo87) [inline wbooks/src/constants/fonts.ts]: FONTS const it's not necessary (the file is called fonts). You can declare a object SIZE and export that.

**Review by mcolombo87 — CHANGES_REQUESTED** [review]

File: wbooks/src/constants/fonts.ts (hunk 8 above)
Comment (CristianRWolox) [reply wbooks/src/constants/fonts.ts]: Done!

File: wbooks/src/app/screens/Library/screens/BookList/index.tsx (hunk 7 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/Library/screens/BookList/index.tsx]: Done!

File: wbooks/src/app/screens/Library/components/BookData/index.tsx (hunk 6 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/Library/components/BookData/index.tsx]: I restructured the branch, now I manage only one screen "Library" (component that contains the FlatList) and within this screen, there is a component name "BookData" (the component that renders the data). In the logic that I propose, there are no more screens inside "Library".

I am attentive to your suggestions.

File: wbooks/src/app/screens/BookData/index.tsx (hunk 5 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/BookData/index.tsx]: Done!

File: wbooks/src/app/screens/BookData/styles.ts (hunk 4 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/BookData/styles.ts]: Done!

**Review by mcolombo87 — APPROVED** [review]
👍

**Review by guidoprinc — APPROVED** [review]

**Pull request merged**